### PR TITLE
lorawan: fix premature return in `lorawan_send`

### PR DESCRIPTION
**Pull request by JordanYates**
Fixes an issue where `lorawan_send` would return prematurely when
`LORAWAN_MSG_CONFIRMED` is mixed with unconfirmed messages.

All calls to `LoRaMacMcpsRequest` result in `McpsConfirm` being run,
where the semaphore `mcps_confirm_sem` is given. However this semaphore
is only taken when `LORAWAN_MSG_CONFIRMED` is set.

Therefore if an unconfirmed message is sent, any following confirmed
messages will return from `lorawan_send` immediately as the semaphore
will be available from the previous send. The return value would also
be wrong for the same reasons.

Fixed by only giving the semaphore in situations when it is being
blocked on.

Signed-off-by: Jordan Yates <jordan.yates@data61.csiro.au>

### CODE REVIEW

File: subsys/lorawan/lorawan.c [hunk 1]
```diff
@@ -85,7 +85,10 @@ static void McpsConfirm(McpsConfirm_t *mcpsConfirm)
 	}
 
 	last_mcps_confirm_status = mcpsConfirm->Status;
-	k_sem_give(&mcps_confirm_sem);
+	/* mcps_confirm_sem is only blocked on in the MCPS_CONFIRMED case */
```
Comment (Mani-Sadhasivam) [inline subsys/lorawan/lorawan.c:88]: Comment could be, ```Release semaphore only for the confirmed frames```

**Review by Mani-Sadhasivam — APPROVED** [review]
Just a nitpick, otherwise LGTM! Thanks for the fix.

File: subsys/lorawan/lorawan.c (hunk 1 above)
Comment (JordanYates) [reply subsys/lorawan/lorawan.c:88]: I don't think the suggested comment is useful, that information can be obtain simply by reading the if statement.
The comment was added to describe why only some calls to `McpsConfirm` give the `mcps_confirm_sem` semaphore.
Having written that, it seems like the name would more appropriately be `confimed_msg_done_sem` or similar.

**Comment by JordanYates** [comment]
> Just a nitpick, otherwise LGTM! Thanks for the fix.

Is there another reviewer that can be added to this PR?
Otherwise I think it will be stuck in 1 approval limbo forever :)

**Review by mniestroj — APPROVED** [review]

**Pull request merged**